pinot-tools/src/main/java/com/linkedin/pinot/tools/admin/command/MergeConfigsCommand.java

```diff
@@ -0,0 +1,689 @@
+      System.out.println("Input directory " + inputDir + " does not exist.");
+      profileConfigurations = List.empty();
+    } else if (!inputDir.isDirectory()) {
+      throw new RuntimeException("Input directory " + inputDir + " is not a directory!");
+    } else {
+      profileConfigurations = List
+          .ofAll(FileUtils.listFiles(profileDir, CONFIG_FILE_EXTENSIONS, true))
+          .map(file -> {
+            Config config = loadConfigFromFile(file);
+            String configName = file.getName().replaceAll("\\.conf$", "");
+
+            return Tuple.of(configName, config);
+          });
+    }
+
+    LOGGER.info("Loaded {} configuration profiles: {}", profileConfigurations.length(), profileConfigurations.map(Tuple2::_1).asJava());
+
+    // Build a map of value to configuration key for each configuration profile
+    Map<TYPE, Map<String, Set<String>>> valueToProfileAndKeys = profileConfigurations
+        .map(profileNameAndConfigTuple -> {
+          String profileName = profileNameAndConfigTuple._1;
+          Config config = profileNameAndConfigTuple._2;
+
+          Seq<Tuple2<TYPE, String>> configValuesAndKeys = HashSet
+              .ofAll(config.entrySet())
+              .toList()
+              .map(entry -> Tuple.of((TYPE) entry.getValue().unwrapped(), entry.getKey()));
+
+          return Tuple.of(profileName, configValuesAndKeys);
+        })
+        .flatMap(profileAndValueKeyTuples -> {
+          String profileName = profileAndValueKeyTuples._1;
+          Seq<Tuple2<TYPE, String>> valuesAndKeys = profileAndValueKeyTuples._2;
+
+          return valuesAndKeys
+              .map(valueAndKeyTuple -> {
+                TYPE value = valueAndKeyTuple._1;
+                String key = valueAndKeyTuple._2;
+
+                return Tuple.of(value, profileName, key);
+              });
+        })
+        .groupBy(valuesProfilesAndKeys -> valuesProfilesAndKeys._1)
+        .map((value, valueProfileKeyTriples) -> {
+          Map<String, Set<String>> profilesAndKeys = valueProfileKeyTriples
+              .groupBy(valueProfileKeyTriple -> valueProfileKeyTriple._2)
+              .map((profile, valueProfileKeyTriples2) -> {
+                Set<String> keys = valueProfileKeyTriples2
+                    .map(valueProfileKeyTriple -> valueProfileKeyTriple._3)
+                    .toSet();
+
+                return Tuple.of(profile, keys);
+              })
+              .toMap(Function.identity());
+
+          return Tuple.of(value, profilesAndKeys);
+        })
+        .toMap(Function.identity())
+        .removeKeys(key -> key.toString().trim().isEmpty());
+
+    // Group input files by their name
+    Map<String, Config> mergedConfigs = configFilesAndProfiles
+        .groupBy(configFileAndProfiles -> configFileAndProfiles._1.getName())
+        .map((configFilename, configFileAndProfilesTuples) -> {
+          System.out.print("\rWorking on " + configFilename + "                            ");
```

Comment:
use logger

Reply:
The `System.out.println` is intentional. Since the merge process can merge a lot of files, the same output line is reused for all the files to show progress as opposed to having one line per file (hence the `\r` at the beginning of the progress string).

Otherwise, if a logger was used, this would mean that if there is an error while doing the merging process, one would have to scroll through all the log entries to figure out the error.